src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -58,21 +60,30 @@ const ProgressiveProfiling = (props) => {
       })
       .catch(() => {});
 
-    let userEnrollmentAction = false;
     if (registrationResponse) {
       setRegistrationResult(registrationResponse);
       sendPageEvent('login_and_registration', 'welcome');
-
-      const queryParams = getAllPossibleQueryParams(registrationResponse.redirectUrl);
-      if ('enrollment_action' in queryParams) {
-        userEnrollmentAction = true;
-      }
     }
+  }, [DASHBOARD_URL, registrationResponse]);
 
-    if (enablePersonalizedRecommendations && !userEnrollmentAction) {
-      setShowRecommendationsPage(true);
+  useEffect(() => {
+    let queryParams = {};
+    let timer = null;
+    if (registrationResponse) {
+      queryParams = getAllPossibleQueryParams(registrationResponse.redirectUrl);
+    }
+    if (enablePersonalizedRecommendations && !('enrollment_action' in queryParams)) {
```

Comment:
This will activate the experiment even if registration response is not present and the condition here will redirect the user to dashboard: https://github.com/openedx/frontend-app-authn/blob/d5513bddecb7a99dc5b9f077091b76a7f32ee676/src/progressive-profiling/ProgressiveProfiling.jsx#L88-L91